rs/state_machine_tests/src/lib.rs

```diff
@@ -1331,11 +1331,22 @@ impl StateMachine {
         StateMachineBuilder::new().with_config(Some(config)).build()
     }
 
+    pub fn execute_round_with_blockmaker_metrics(&self, blockmaker_metrics: BlockmakerMetrics) {
+        let mut payload = self.build_payload();
+        payload = payload.with_blockmaker_metrics(blockmaker_metrics);
+        self.execute_payload(payload);
+    }
+
+    pub fn execute_round(&self) {
+        let payload = self.build_payload();
+        self.execute_payload(payload);
+    }
+
     /// Assemble a payload for a new round using `PayloadBuilderImpl`
     /// and execute a round with this payload.
     /// Note that only ingress messages submitted via `Self::submit_ingress`
     /// will be considered during payload building.
-    pub fn execute_round(&self) {
+    fn build_payload(&self) -> PayloadBuilder {
```

Comment:
The name of this function is misleading as it also has side-effects, e.g., certifying the latest state, beyond bulding the payload. Could you please follow my initial recommendation:
```
pub fn execute_round(&self) {
  self.do_execute_round(None);
}
pub fn execute_round_with_blockmaker_metrics(&self, blockmaker_metrics: BlockmakerMetrics) {
  self.do_execute_round(Some(blockmaker_metrics));
}
pub fn do_execute_round(&self, blockmaker_metrics: Option<BlockmakerMetrics>) {
   // the original implementation
  if let Some(blockmaker_metrics) = blockmaker_metrics {
    payload = payload.with_blockmaker_metrics(blockmaker_metrics);
  }
  self.execute_payload(payload);
}
```